packages/client/src/Components/Mapping/helpers/planningRouteFor.js

```diff
@@ -56,55 +63,161 @@ function bearingBetween (/* latLng */ p1, /* latLng */ p2) {
  */
 function bearingMarkerFor (/* number */ angle) {
   let icon
+  var iconAngles = {
+    turn0degVar: {icon: turn0deg, markerAngle: 0},
+    turn30degVar: {icon: turn30deg, markerAngle: 30},
+    turn60degVar: {icon: turn60deg, markerAngle: 60},
+    turn90degVar: {icon: turn90deg, markerAngle: 90},
+    turn120degVar: {icon: turn120deg, markerAngle: 120},
+    turn150degVar: {icon: turn150deg, markerAngle: 150},
+    turnLightVar: {icon: lightTurn, markerAngle: null},
+    turnBackVar: {icon: backTurn, markerAngle: null},
+  }
 
-  if (angle === 180 || angle === 360 || angle === 0) {
-    // for the 180 and 360, which are flat icons so end icon is used
-    icon = turnEnd
-  } else if (angle > 0 && angle < 90) {
-    // North to East
-    icon = turnLeft
-  } else if (angle < 180) {
-    // East to South
-    icon = turnRight
+  // console.log(iconAngles.turn0degVar)
+  // console.log("Bearing angle", angle)
+  if(angle === 0 ||  angle == 89 ||angle === 180 || angle === 360){
```

Comment:
the '89' looks out of place here. The other values are all north/south.  But 89 is close to 'East'.    Since it's not on a 90 degree value, I suspect it's not matching anything.

Reply:
Oh, this is a horrible if structure.  I suggest you round the angle to the nearest 30 degrees - it should really simplify things.

Reply:
There's a function in the repo called `roundToNearest()` . You'd just call `angle = roundToNearest(angle, 30)`